### PR TITLE
v7.0.0-beta.5

### PR DESCRIPTION
**Pull request by romgrk**
### Release the packages

- [x] Checkout the last version of the working branch
- [x] Make sure you have the latest dependencies installed: `yarn`.
- [ ] Build the packages: `yarn release:build`.
- [ ] Release the versions on npm: `yarn release:publish` (you need your 2FA device).
- [ ] Push the newly created tag: `yarn release:tag`.

### Publish the documentation

The documentation must be updated on the `docs-vX` branch (`docs-v4` for `v4.X` releases, `docs-v5` for `v5.X` releases, ...)

- [ ] Push the working branch on the documentation release branch to deploy the documentation with the latest changes.

```bash
git push upstream next:docs-next -f
```

You can follow the deployment process [on the Netlify Dashboard](https://app.netlify.com/sites/material-ui-x/deploys?filter=docs-v6)
Once deployed, it will be accessible at https://material-ui-x.netlify.app/ for the `docs-v6` deployment.

### Publish GitHub release

- [ ] After documentation is deployed, publish a new release on [GitHub releases page](https://github.com/mui/mui-x/releases)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,85 @@
...
+
+- [DataGridPremium] Make clipboard copy respect the sorting during cell selection (#12235) @MBilalShafi
+
+### Date Pickers
+
+#### `@mui/x-date-pickers@7.0.0-beta.5`
+
+- [pickers] Fix toolbar components props handling (#12211) @LukasTy
+- [DateTimeRangePicker] Fix validation behavior (#12243) @LukasTy
+- [I10n] Improve Chinese (zh-CN) locale (#12245) @headironc
```
Comment (LukasTy) [inline CHANGELOG.md:53]:
```suggestion
- [l10n] Improve Chinese (zh-CN) locale (#12245) @headironc
- [l10n] Improve Korean (ko-KR) locale (#12192) @Luzi
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,85 @@
...
+Same changes as in `@mui/x-data-grid-pro@7.0.0-beta.5`, plus:
+
+- [DataGridPremium] Make clipboard copy respect the sorting during cell selection (#12235) @MBilalShafi
+
+### Date Pickers
+
+#### `@mui/x-date-pickers@7.0.0-beta.5`
+
+- [pickers] Fix toolbar components props handling (#12211) @LukasTy
+- [DateTimeRangePicker] Fix validation behavior (#12243) @LukasTy
```
Comment (LukasTy) [inline CHANGELOG.md:52]: Need to update the script 🙈 
```suggestion
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,85 @@
...
+- [I10n] Improve Chinese (zh-CN) locale (#12245) @headironc
+
+#### `@mui/x-date-pickers-pro@7.0.0-beta.5` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')
+
+Same changes as in `@mui/x-date-pickers@7.0.0-beta.5`.
```
Comment (LukasTy) [inline CHANGELOG.md:57]:
```suggestion
Same changes as in `@mui/x-date-pickers@7.0.0-beta.5`.

- [DateTimeRangePicker] Fix validation behavior (#12243) @LukasTy
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,85 @@
...
+- [docs] Fix image layout shift when loading @oliviertassinari
+- [docs] Match Material UI repo comment for redirections @oliviertassinari
+- [docs] Non breaking spaces @oliviertassinari
+- [docs] Standardize WAI-ARIA references @oliviertassinari
+
+### Core
+
+- [core] Allow local docs next.js settings (#12227) @romgrk
+- [core] Remove grid folder from `getComponentInfo` RegExp (#12241) @flaviendelangle
+- [core] Remove window. reference for common globals @oliviertassinari
```
Comment (LukasTy) [inline CHANGELOG.md:78]:
```suggestion
- [core] Remove `window.` reference for common globals @oliviertassinari
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,85 @@
...
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 7.0.0-beta.5
+
+_Feb 29, 2024_
+
+We'd like to offer a big thanks to the 13 contributors who made this release possible. Here are some highlights ✨:
+
+- 🎁 Add `getSortComparator` for more advanced sorting behaviors (#12215) @cherniavskii
+- 🚀 Add `use client` directive to the Grid packages (#11803) @MBilalShafi
```
Comment (LukasTy) [inline CHANGELOG.md:13]:
```suggestion
- 🚀 Add `use client` directive to the Grid packages (#11803) @MBilalShafi
- 🌍 Improve Korean (ko-KR) and Chinese (zh-CN) locales on the Pickers
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -2352,6 +2431,37 @@ Here is an example of the renaming for the `<ChartsTooltip />` component.
...
 - [license] Correctly throw errors (#10924) @oliviertassinari
 
+## 6.19.6
+
+_Feb 29, 2024_
+
+We'd like to offer a big thanks to the 3 contributors who made this release possible. Here are some highlights ✨:
+
+- 🐞 Bugfixes
+- 🌍 Improve Korean (ko-KR) locale on the Data Grid (#11972) @Luzi
```
Comment (LukasTy) [inline CHANGELOG.md:2441]: Nitpick: We need to update these v6 changes as well after changes in the respective PR. 👌

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,85 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 7.0.0-beta.5
+
+_Feb 29, 2024_
+
+We'd like to offer a big thanks to the 13 contributors who made this release possible. Here are some highlights ✨:
```
Comment (LukasTy) [inline CHANGELOG.md:10]: If I'm not mistaken counting. 🙈 
```suggestion
We'd like to offer a big thanks to the 14 contributors who made this release possible. Here are some highlights ✨:
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,87 @@
...
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 7.0.0-beta.5
+
+_Feb 29, 2024_
+
+We'd like to offer a big thanks to the 14 contributors who made this release possible. Here are some highlights ✨:
+
+- 🎁 Add `getSortComparator` for more advanced sorting behaviors (#12215) @cherniavskii
+- 🚀 Add `use client` directive to the Grid packages (#11803) @MBilalShafi
```
Comment (cherniavskii) [inline CHANGELOG.md:13]:
```suggestion
- 🚀 Add `use client` directive to the Data Grid packages (#11803) @MBilalShafi
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,6 +3,87 @@
...
+### Data Grid
+
+#### `@mui/x-data-grid@7.0.0-beta.5`
+
+- [DataGrid] Add `getSortComparator` for more advanced sorting behaviors (#12215) @cherniavskii
+- [DataGrid] Add `use client` directive to the Grid packages (#11803) @MBilalShafi
+- [DataGrid] Fix `disableResetButton` and `disableShowHideToggle` flags to not exclude each other (#12169) @adyry
+- [DataGrid] Fix cell range classnames (#12230) @romgrk
+- [DataGrid] Fix wrong offset for right-pinned columns when toggling dark/light modes (#12233) @cherniavskii
+- [DataGrid] Refactor: row virtualization & rendering (#12247) @romgrk
```
Comment (cherniavskii) [inline CHANGELOG.md:27]:
```suggestion
- [DataGrid] Improve row virtualization and rendering performance (#12247) @romgrk
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -3,6 +3,87 @@
...
+- [DataGrid] Fix boolean "is" filter (#12117) @shaharyar-shamshi
+- [DataGrid] Fix `upsertFilterItems` removing filters that are not part of the update (#11954) @gitstart
+
+#### `@mui/x-data-grid-pro@7.0.0-beta.5` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')
+
+Same changes as in `@mui/x-data-grid@7.0.0-beta.5`, plus:
+
+- [DataGridPro] Fix column resize errors on MacOS with automatic scrollbars enabled (#12217) @cherniavskii
+- [DataGridPro] Fix lazy-loading crash (#12080) @romgrk
+- [DataGridPro] Fix useGridRows not giving error on reversed data (#10821) @martijn-basesoft
```
Comment (cherniavskii) [inline CHANGELOG.md:39]:
```suggestion
- [DataGridPro] Fix lazy loading error when existing rows are passed to `replaceRows`
```

**Review by cherniavskii — APPROVED** [review]

File: CHANGELOG.md [hunk 11]
```diff
@@ -3,6 +3,87 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 7.0.0-beta.5
+
+_Feb 29, 2024_
```
Comment (LukasTy) [inline CHANGELOG.md:8]:
```suggestion
_Mar 1, 2024_
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -2352,6 +2433,36 @@ Here is an example of the renaming for the `<ChartsTooltip />` component.
 - [core] Update release instructions as per v7 configuration (#10962) @MBilalShafi
 - [license] Correctly throw errors (#10924) @oliviertassinari
 
+## 6.19.6
+
+_Feb 29, 2024_
```
Comment (LukasTy) [inline CHANGELOG.md:2438]:
```suggestion
_Mar 1, 2024_
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -2352,6 +2433,36 @@ Here is an example of the renaming for the `<ChartsTooltip />` component.
...
+We'd like to offer a big thanks to the 4 contributors who made this release possible. Here are some highlights ✨:
+
+- 🌍 Improve Korean (ko-KR) and Chinese (zh-CN) locales on the Pickers
+- 🐞 Bugfixes
+
+### Data Grid
+
+#### `@mui/x-data-grid@6.19.6`
+
+- [l10n] Improve Korean (ko-KR) locale (#12186) @Luzi
```
Comment (LukasTy) [inline CHANGELOG.md:2449]:
```suggestion
No changes.
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -2352,6 +2433,36 @@ Here is an example of the renaming for the `<ChartsTooltip />` component.
...
+
+#### `@mui/x-data-grid-pro@6.19.6` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')
+
+Same changes as in `@mui/x-data-grid@6.19.6`.
+
+#### `@mui/x-data-grid-premium@6.19.6` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')
+
+Same changes as in `@mui/x-data-grid-pro@6.19.6`, plus:
+
+- [DataGridPremium] Make clipboard copy respect the sorting during cell selection (#12255) @MBilalShafi
```
Comment (LukasTy) [inline CHANGELOG.md:2459]:
```suggestion
- [DataGridPremium] Make clipboard copy respect the sorting during cell selection (#12255) @MBilalShafi

### Date Pickers

#### `@mui/x-date-pickers@6.19.6`

- [l10n] Improve Chinese (zh-CN) locale (#12250) @headironc
- [l10n] Improve Korean (ko-KR) locale (#12186) @Luzi

#### `@mui/x-date-pickers-pro@6.19.6` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@6.19.6`.
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -2352,6 +2433,36 @@ Here is an example of the renaming for the `<ChartsTooltip />` component.
...
+
+### Docs
+
+- [docs] Update lazy loading demo to show skeleton rows during initial rows fetch (#12062) @cherniavskii
+- [l10n] Improve Chinese (zh-CN) locale (#12250) @headironc
```
Comment (LukasTy) [inline CHANGELOG.md:2464]:
```suggestion
```

File: CHANGELOG.md [hunk 16]
```diff
@@ -2352,6 +2433,36 @@ Here is an example of the renaming for the `<ChartsTooltip />` component.
...
 - [license] Correctly throw errors (#10924) @oliviertassinari
 
+## 6.19.6
+
+_Feb 29, 2024_
+
+We'd like to offer a big thanks to the 4 contributors who made this release possible. Here are some highlights ✨:
+
+- 🌍 Improve Korean (ko-KR) and Chinese (zh-CN) locales on the Pickers
+- 🐞 Bugfixes
```
Comment (LukasTy) [inline CHANGELOG.md:2443]:
```suggestion
- 🐞 Bugfixes
- 📚 Documentation improvements
```

File: CHANGELOG.md [hunk 17]
```diff
@@ -3,6 +3,87 @@
...
+
+### Tree View / `@mui/x-tree-view@7.0.0-beta.5`
+
+- [TreeView] Add public API and expose focus method (#12143) @noraleonte
+
+### Docs
+
+- [docs] Fix image layout shift when loading @oliviertassinari
+- [docs] Match Material UI repo comment for redirections @oliviertassinari
+- [docs] Non breaking spaces @oliviertassinari
```
Comment (LukasTy) [inline CHANGELOG.md:73]:
```suggestion
- [docs] Non breaking spaces @oliviertassinari
- [docs] Polish the Date Picker playground (#11869) @zanivan
```

File: CHANGELOG.md [hunk 18]
```diff
@@ -3,6 +3,87 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 7.0.0-beta.5
+
+_Feb 29, 2024_
+
+We'd like to offer a big thanks to the 14 contributors who made this release possible. Here are some highlights ✨:
```
Comment (LukasTy) [inline CHANGELOG.md:10]:
```suggestion
We'd like to offer a big thanks to the 15 contributors who made this release possible. Here are some highlights ✨:
```

**Review by LukasTy — APPROVED** [review]

**Pull request merged**